### DISCUSSION TITLE
Being able to move the node_modules directory (like it's possible in yarn(JS), and in composer (PHP))

### DISCUSSION THREAD

**Discussion by bakeiro** [npm/rfcs#265] [General]
For reasons like convenience, design, or special cases (like setting up a shared folder containing the JS dependencies), would be nice to be able to specify the node_modules folder location **locally** to the project, and not globally.

For example, you can achieve easily this in composer writing `"vendor-dir": "./custom_folder_name/vendor/"` in composer.json

**Comment by ljharb** [comment]
That would break any tooling that expects `node_modules` to be there, including anything that does module resolution (linters, bundlers, etc) - that doesn't feel very convenient to me.

**Reply by bakeiro** [reply]
yes that makes sense, but NODE_PATH global variable it's able to do this already, but for all packages globally so it's already more less possible but not only local, still, I get your point, since it would be a big breaking change

**Reply by ljharb** [reply]
NODE_PATH is largely deprecated and highly discouraged, and doesn't exist in the ESM implementation, so I don't think this changes the discussion much.

**Reply by bakeiro** [reply]
yeah, I guess that makes sense... marked as answered ;) (but not 100% satisfied haha 😄  )
